tensorflow/core/api_def/base_api/api_def_DiagPart.pbtxt

```diff
@@ -22,15 +22,22 @@ tensor of rank `k` with dimensions `[D1,..., Dk]` where:
 
 `diagonal[i1,..., ik] = input[i1, ..., ik, i1,..., ik]`.
 
-For example:
+For a rank 2 tensor, `diag_part` and `tensor_diag_part`
+produce the same result. An example where they differ
+is given below.
 
 ```
-# 'input' is [[1, 0, 0, 0]
-              [0, 2, 0, 0]
-              [0, 0, 3, 0]
-              [0, 0, 0, 4]]
+# 'input' is [[[[1111,1112],[1121,1122]],
+               [[1211,1212],[1221,1222]]],
+              [[[2111, 2112], [2121, 2122]],
+               [[2211, 2212], [2221, 2222]]]
+             ]
 
-tf.diag_part(input) ==> [1, 2, 3, 4]
+tf.tensor_diag_part(input) ==> [[1111, 1212], [2121, 2222]]
+tf.diag_part(input) ==> [[[1111, 1122],
+                          [1211, 1222]],
+                         [[2111, 2122],
+                          [2211, 2222]]]
```

Comment:
Can you rewrite this using testable docstrings please?

https://www.tensorflow.org/community/contribute/docs_ref#testable_docstrings

Reply:
given that this is not a python file, I wasn't sure whether python syntax would be appropriate here, so I kept it in the old format. In an old PR for the same issue, you've linked https://github.com/tensorflow/tensorflow/commit/8582a583d45fdf6a895c0677c7f68b4bf3596ee9 as an example for adding python docstrings. Unfortunately, I'm not familiar enough about how the python API is constructed in the end to make sure to do this correctly. 
For example, where is the `@dispatch.add_dispatch_support` support in the linked commit coming from?

Reply:
https://github.com/tensorflow/tensorflow/blob/v2.3.0/tensorflow/python/ops/array_ops.py#L2449-L2587 is the python docstring for the op. You should not need to do the same as 8582a58 since we already have the python binding

Reply:
I think you are confusing exactly the two methods I mention in this pull request. The method you refer to is `matrix_diag_part`, whereas the method I'm working with here is `tensor_diag_part`. These methods differ when tensors of `rank > 2` are concerned. In fact, the very first example of the method you linked shows a tensor of shape `(2, 3, 4)`, which would not be a valid input for `tensor_diag_part`, which requires its input to be "Rank k tensor where k is even and not zero."

Reply:
You're right, my bad. You should then expose a Python entry-point like in `8582a58`. Copy the implementation from `matrix_diag_part`, just replacing `matrix` parts with `tensor` ones.

The decorator comes from an import, see https://github.com/tensorflow/tensorflow/blob/b36436b087bd8e8701ef51718179037cccdfc26e/tensorflow/python/ops/array_ops.py#L44